OrbitGl/App.cpp

```diff
@@ -1522,23 +1523,55 @@ void OrbitApp::DeselectTracepoint(const TracepointInfo& tracepoint) {
   return data_manager_->IsTracepointSelected(info);
 }
 
+void OrbitApp::EnableFrameTrack(const FunctionInfo& function) {
+  data_manager_->EnableFrameTrack(function);
+}
+
+void OrbitApp::DisableFrameTrack(const FunctionInfo& function) {
+  data_manager_->DisableFrameTrack(function);
+}
+
 void OrbitApp::AddFrameTrack(const FunctionInfo& function) {
-  GetMutableCaptureData().InsertFrameTrack(function);
-  data_manager_->set_user_defined_capture_data(GetCaptureData().user_defined_capture_data());
-  AddFrameTrackTimers(function);
+  if (!HasCaptureData()) {
+    return;
+  }
+  const uint64_t function_address = GetCaptureData().GetAbsoluteAddress(function);
+  if (GetCaptureData().GetSelectedFunction(function_address) == nullptr) {
+    return;
+  }
+  // We only add a frame track to the actual capture data if the function for the frame
+  // track actually has hits in the capture data. Otherwise we can end up in inconsistent
+  // states where "empty" frame tracks exist in the capture data (which would also be
+  // serialized).
+  const FunctionStats& stats = GetCaptureData().GetFunctionStatsOrDefault(function);
+  if (stats.count() > 1) {
+    GetMutableCaptureData().EnableFrameTrack(function);
+    AddFrameTrackTimers(function);
+  } else {
+    CHECK(empty_frame_frame_warning_callback_);
+    empty_frame_frame_warning_callback_();
```

Comment:
-> `empty_frame_track_warning_callback_`

Reply:
Thanks for spotting. Fixed.